plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageLayoutHelper.java

```diff
@@ -379,6 +381,7 @@ public static String getLegacyDatastorePathFromVmdkFileName(DatastoreMO dsMo, St
 
     //This method call is for the volumes to be created or can also be for volumes already exists
     public static String getDatastorePathBaseFolderFromVmdkFileName(DatastoreMO dsMo, String vmdkFileName) throws Exception {
+        HypervisorHostHelper.createBaseFolderInDatastore(dsMo, dsMo.getDataCenterMor());
         return String.format("[%s] %s/%s", dsMo.getName(), HypervisorHostHelper.VSPHERE_DATASTORE_BASE_FOLDER, vmdkFileName);
```

Comment:
@harikrishna-patnala changes LGTM, didn't test. Hope all the datastores are covered, and 'fcd' folder is created. Please run the smoke tests and check if there are any issues.